recipes-app/iot2050-firmware-update/files/iot2050-firmware-update

```diff
@@ -9,6 +9,30 @@
 # This file is subject to the terms and conditions of the MIT License.  See
 # COPYING.MIT file in the top-level directory.
 #
+# To use this tool, a <firmware-update-package>.tar.xz and its sha256 checksum file
+# <firmware-update-package>.tar.xz.sha256 are needed.
+#
+# The <firmware-update-package>.tar.xz contains below files:
+#   - firmware.bin: The firmware to update.
+#   - firmware.bin.sha256: The sha256 checksum of the firmware.
+#   - update.conf.json: The update criteria. Currently only the OS image version
+#       is checked.
+#
+# Example of update.conf.json:
+# {
+#     "update_criteria":[
+#         {
+#             "image_type": "Example Image",
+#             "version_field": "BUILD_ID",
+#             "version_value": "V01.01.00"
```

Comment:
What's the semantic of this version field? Exact match? Or "greater or equal"?

Reply:
startswith

Reply:
Then let's call it 'min_version'.

Reply:
python startswith, so maybe "version_prefix"

Reply:
OK, then this is kind of an exact match - why?

Reply:
In some cases we need to cover more than 1 minor versions, for example, we need to cover both 1.1.1 and 1.1.2, then we could use 1.1 as the match field.

Other way is to use the standard regex, maybe that's a better solution?

Reply:
No, I mean why not using a min_version semantic? Without that, we may eventually have to release a new firmware package just declare it being compatible with installing it over a newer image version, without actually having to change its code for that.

Reply:
That could be the case. Do you think to cover min_version semantic is enough here?

Reply:
It's a trade-off, true: Theoretically, a firmware version we release tomorrow could become incompatible with an image version we release later on. But, unless someone breaks the SYSFW ABI again, this may be rather unlikely from now on.

If we see a remaining risk, we would indeed have to define an upper limit of supported image versions already today, say `< V02.*`. But what if we feel like bumping the image version to a new major level for other reasons, e.g. due to a distro update? Then you also logically break compatibility with existing firmware update packages.